hail/python/hail/methods/statgen.py

```diff
@@ -1983,6 +1998,428 @@ def _linear_skat(group,
     return ht.select_globals('y_residual', 's2', 'n_complete_samples')
 
 
+@typecheck(group=expr_any,
+           weight=expr_float64,
+           y=expr_float64,
+           x=expr_float64,
+           covariates=sequenceof(expr_float64),
+           max_size=int,
+           null_max_iterations=int,
+           null_tolerance=float,
+           accuracy=numeric,
+           iterations=int)
+def _logistic_skat(group,
+                   weight,
+                   y,
+                   x,
+                   covariates,
+                   max_size: int = 46340,
+                   null_max_iterations: int = 25,
+                   null_tolerance: float = 1e-6,
+                   accuracy: float = 1e-6,
+                   iterations: int = 10000):
+    r'''The logistic sequence kernel association test (SKAT).
+
+    Logistic SKAT tests if the phenotype, `y`, is significantly associated with the genotype,
+    `x`. For :math:`N` samples, in a group of :math:`M` variants, with :math:`K` covariates, the
+    model is given by:
+
+    .. math::
+
+        \begin{align*}
+        X &: R^{N \times K} \\
+        G &: \{0, 1, 2\}^{N \times M} \\
+        \\
+        \varepsilon &\sim N(0, \sigma^2) \\
+        \textrm{logit}(P(y=1)) &= \beta_0 X + \beta_1 G + \varepsilon
+        \end{align*}
+
+    The usual null hypothesis is :math:`\beta_1 = 0`. SKAT tests for an association, but does not
+    provide an effect size or other information about the association.
+
+    Wu et al. argue that, under the null hypothesis, a particular value, :math:`Q`, is distributed
+    according to a generalized chi-squared distribution with parameters determined by the genotypes,
+    weights, and residual phenotypes. The SKAT p-value is the probability of drawing even larger
+    values of :math:`Q`. :math:`Q` is defined by Wu et al. as:
+
+    .. math::
+
+        \begin{align*}
+        r &= y - \textrm{logit}^{-1}(\widehat{\beta_\textrm{null}} X) \\
+        W_{ii} &= w_i \\
+        \\
+        Q &= r^T G W G^T r
+        \end{align*}
+
+    :math:`\widehat{\beta_\textrm{null}}` is the best-fit beta under the null model:
+
+    .. math::
+
+        \textrm{logit}(P(y=1)) = \beta_\textrm{null} X + \varepsilon \quad\quad \varepsilon \sim N(0, \sigma^2)
+
+    Therefore :math:`r`, the residual phenotype, is the portion of the phenotype unexplained by the
+    covariates alone. Also notice:
+
+    1. The residual phenotypes are Bernoulli distributed with mean :math:`p` and variance
+       :math:`\sigma^2 = p(1 - p)` where :math:`p` is the best-fit probability.
+
+    2. :math:`G W G^T`, is a symmetric positive-definite matrix when the weights are non-negative.
+
+    We can transform the residuals into standard normal variables by normalizing by their
+    variance.
```

Comment:
I think everything below this doesn't make much sense. The residuals are certainly not normal -- they can only take two possible values. I tried to look into how to justify the logistic case, and made a little progress ("Pearson residuals" are relevant), but I think it's best to not try to write a complete derivation here. It should be enough to summarize the math from the paper that we implement.